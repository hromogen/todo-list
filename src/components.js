import React from 'react';
import { theStore } from './reducers';

const { Component } = React;
export const Counter = ({value, onIncrement, onDecrement}) => {
    return (
        <article className="counter">
            <h1>{value}</h1>
            <button onClick={onIncrement}>+</button>
            <button onClick={onDecrement}>-</button>
        </article>
    )
};

let todoCounter = 0;
export class TodoApp extends Component {
    constructor(props){
        super(props);
        this.state = {
            value: ''
        };
    }
    onSubmit(event) {
        event.preventDefault();
        theStore.dispatch({
            type: 'ADD_TODO',
            text: this.input.value,
            id: todoCounter++
        });
        this.input.value = '';
    };
    render(){
        return (
            <article className="todoList">
                <form action="" onSubmit={event => this.onSubmit(event)}>
                    <input ref={node => {this.input = node}} type="text"/>
                    <button type="submit">Add task</button>
                </form>
                <ul>
                    {
                        this.props.todos &&
                        this.props.todos.length &&
                        this.props.todos.map(todo => todo.text ? <li key={todo.id}>
                            {todo.text}
                            </li> : ""
                        )
                    }
                </ul>
            </article>
        );
    };
};