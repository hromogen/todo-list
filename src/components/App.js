import React from 'react';
import Footer from './Footer';
import AddTodo from '../containers/AddTodo';
import VisibleTodoList from '../containers/VisibleTodoList';
 
const App = () => (
  <main>
    <AddTodo />
    <VisibleTodoList />
    <Footer />
  </main>
)
 
export default App