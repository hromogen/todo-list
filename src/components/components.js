import React from 'react';
import { theStore } from './reducers';

const { Component } = React;
export const Counter = ({value, onIncrement, onDecrement}) => {
    return (
        <article className="counter">
            <h1>{value}</h1>
            <button onClick={onIncrement}>+</button>
            <button onClick={onDecrement}>-</button>
        </article>
    )
};

let todoCounter = 0;

const FilterLink = ({filter, children}) => {
    return (
        <a href="#"
           onClick={e => {
               e.preventDefault();
               theStore.dispatch({
                   type: 'SET_VISIBILTY_FILTER',
                   filter })
           }
           }>
            {children}
        </a>
    );
};

const getVisibleTodos = (todos, filter) => {
    console.log('todos ', todos);
    console.log('filter ', filter);
    switch(filter){
        case 'SHOW_ALL':
            return todos;
        case 'SHOW_COMPLETED':
            return todos.filter(t => t.completed);
        case 'SHOW_ACTIVE':
            return todos.filer(t => !t.completed);
    }
};

export class TodoApp extends Component {
    onSubmit(event) {
        event.preventDefault();
        theStore.dispatch({
            type: 'ADD_TODO',
            text: this.input.value,
            id: todoCounter++
        });
        this.input.value = '';
    };
    render(){
        const visibleTodos = getVisibleTodos(
            this.props.todos,
            this.props.visibilityFilter
        );
        return (
            <article className="todoList">
                <form action="" onSubmit={event => this.onSubmit(event)}>
                    <input ref={node => {this.input = node}} type="text"/>
                    <button type="submit">Add task</button>
                </form>
                <ul>
                    {
                        visibleTodos.map(
                            todo => todo.text ?
                                <li key={todo.id}
                                onClick={(event) => {
                                    event.preventDefault();
                                    theStore.dispatch({
                                        type: 'TOGGLE_TODO',
                                        id: todo.id
                                    })
                                }}
                                style={{textDecoration: todo.completed ? 'line-through' : 'none'}}>
                                    {todo.text}
                                </li> : "\n"
                            )
                    }
                </ul>
                <p>
                    Show:
                    {'  '}
                    <FilterLink filter='SHOW_ALL'>
                        All
                    </FilterLink>
                    {'  '}
                    <FilterLink filter='SHOW_ACTIVE'>
                        Active
                    </FilterLink>
                    {'  '}
                    <FilterLink filter='SHOW_COMPLETED'>
                        Completed
                    </FilterLink>
                </p>
            </article>
        );
    };
};