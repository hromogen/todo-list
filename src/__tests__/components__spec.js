import React from 'react';

import { TodoApp } from '../components';
import { theStore } from '../reducers';

import Enzyme, { render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
describe('todoApp', () => {
    it('should render correctly', () => {
        const output = shallow(<TodoApp/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('changes state when got text inputet', done => {
        const wrapper = shallow(<TodoApp todos = {theStore.getState().todos}/>),
            input = wrapper.find('input'),
            form = wrapper.find('form');
        input.simulate('change', { target: { value: 'Hello' } });
        form.simulate('submit');
        expect(form.find('li').text()).toBe('Hello');
    });
});
