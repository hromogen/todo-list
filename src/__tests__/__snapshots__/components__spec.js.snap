// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoApp should render correctly 1`] = `
<article
  className="todoList"
>
  <form
    action=""
    onSubmit={[Function]}
  >
    <input
      type="text"
    />
    <button
      type="submit"
    >
      Add task
    </button>
  </form>
  <ul />
</article>
`;
